common/src/main/java/apoc/util/FileUtils.java

```diff
@@ -49,13 +50,26 @@
 import org.apache.commons.io.FilenameUtils;
 import org.apache.commons.lang3.StringUtils;
 import org.neo4j.graphdb.security.URLAccessChecker;
+import org.neo4j.graphdb.security.URLAccessValidationError;
 
 /**
  * @author mh
  * @since 22.05.16
  */
 public class FileUtils {
 
+    public static String getLoadFileUrl(String fileName, URLAccessChecker urlAccessChecker)
+            throws MalformedURLException, URLAccessValidationError {
+        URL url;
+        try {
+            url = URI.create(fileName).toURL();
```

Comment:
So with this approach we take a file, turn it into a URL, turn it back to a string, then back to a URL? maybe we could have this combined in the checkAllowedUrlAndPinToIP() function instead?